fedimint-client/src/lib.rs

```diff
@@ -1494,6 +1494,24 @@ impl ClientBuilder {
         }
     }
 
+    pub async fn get_federation_id(&self) -> anyhow::Result<FederationId> {
+        if let Some(db) = self.db.as_ref() {
+            let db = match db {
```

Comment:
We should have a method on `DatabaseSource` for it.

Reply:
We should just get rid of these DB source shenanigans next release. They are only useful if one can't re-open the same DB because the client leaked a DB ref somewhere. I think it was related to recovery where the DB is closed and reopened without restarting the process.